addOns/pscanrulesAlpha/src/main/java/org/zaproxy/zap/extension/pscanrulesAlpha/JsoScanner.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2019 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.pscanrulesAlpha;
+
+import java.net.*;
+import java.nio.charset.*;
+import java.util.*;
+import net.htmlparser.jericho.*;
+import org.parosproxy.paros.*;
+import org.parosproxy.paros.core.scanner.*;
+import org.parosproxy.paros.network.*;
+import org.zaproxy.zap.extension.pscan.*;
+
+/** Java Serialized Objects (JSO) scanner. Detect the magic sequence and generate an alarm */
+public class JsoScanner extends PluginPassiveScanner {
+
+    /** Prefix for internationalized messages used by this rule */
+    private static final String MESSAGE_PREFIX = "pscanalpha.jso.";
+
+    private static final byte[] JSO_BYTE_MAGIC_SEQUENCE = {(byte) 0xac, (byte) 0xed, 0x00, 0x05};
+    private static final String JSO_BASE_64_MAGIC_SEQUENCE = "rO0AB";
+    private PassiveScanThread parent;
+
+    @Override
+    public void scanHttpRequestSend(HttpMessage msg, int id) {
+        // do nothing
+    }
+
+    @Override
+    public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
+        checkJsoInHeaders(msg);
+
+        checkJsoInCookies(msg);
+
+        checkJsoInBody(msg);
```

Comment:
Think it might be worth doing this in scanHttpRequestSend as well? In case Jso is being POSTed, etc.

The method could be changed from using the msg to accepting the body.

Reply:
You mean something like: `checkJsoInBody(HttpMessage msg, HttpResponseBody responseBody)` ?
HttpMessage is used to build the alarm.

Reply:
No I mean if you're also going to use it in the scanHttpRequestSend method (if that suggestion makes sense) then `checkJsoInBody(String body)` or `checkJsoInBody(Source Body)` and appropriately pass the request or response body.